src/Symfony/Component/FeatureFlags/Attribute/AsStrategy.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symfony\Component\FeatureFlags\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+final class AsStrategy
+{
+    public function __construct(
+        public readonly string $feature,
+        public readonly bool $default = false,
```

Comment:
on pourra aussi rajouter description of course

Reply:
Est-ce que le default est utile? Si je déclare une classe avec sa propre stratégie, retourner Abstain équivaut à retourner default.
J'ai hésité à autoriser le StartegyResult dans ma PR mais, à l'usage, seul le bool est utilisé.

Reply:
du point de vue de l'attribute je suis d'accord il est pas tres pertinent. En revanche le CallbackStrategy a du sens meme en dehors de l'attribute du coup je pense que ca fait sens de garder le StratResult. et a cause de ca il faudrait que l'attribute expose un default car on pourra pas empecher qu'ils return un StratResult aussi via l'attribute.

Reply:
Du coup, on peut le retirer du `#[AsStrategy]` mais le garder dans le `CallbackStrategy` ?

Reply:
on peut oui... je vais voir ce que ca donne

Reply:
pushed. pour etre honnete je suis pas fan. Je trouve que ca crée une incoherence entre la possibilité d'usage et le typage. meme si en effet le rendre optionnel aurait pas beaucoup plus de sens... dans le debug le default apparaitra alors que en vrai du coup il aurait pas de default value. du coup ca devient un peu chelou non ? Bon en tout cas c'est pushed

Reply:
Quoiqu'il arrive, le default dans le Debug est chelou car on ne va vraisemblablement pas le définir. J'ai pas d'exemple en tête où on va utiliser l'attribute et une valeur par défaut.

Ça va relancer le débat du bool vs. enum 🫤

Reply:
indeed. de toute facon on est clair sur le sujet : le abstain n'a de sens que lorsqu'on nest les strats ou qu'on veut plus de granularité sur ce qui se passe. mais dans le cas d'une strat one shot pour une ou 2 features alors un simple bool fera l'affaire.